src/context/__tests__/index.ts

```diff
@@ -36,6 +37,68 @@ describe('Context class', () => {
     );
   });
 
+  test('should throw an error if accessing the harvester query submodule without an active connection', async () => {
+    const newPair = {
+      address: 'someAddress1',
+      meta: {},
+      publicKey: 'publicKey',
+    };
+    dsMockUtils.configureMocks({
+      keyringOptions: {
+        addFromSeed: newPair,
+      },
+    });
+    dsMockUtils.createQueryStub('identity', 'keyToIdentityIds', {
+      returnValue: dsMockUtils.createMockOption(
+        dsMockUtils.createMockLinkedKeyInfo({
+          Unique: dsMockUtils.createMockIdentityId('someDid'),
+        })
+      ),
+    });
+
+    const context = await Context.create({
+      polymeshApi: dsMockUtils.getApiInstance(),
+      harvesterClient: null,
+      seed: 'Alice'.padEnd(32, ' '),
+    });
+
+    expect(() => context.harvesterClient.query).toThrow(
+      'Cannot perform this action without an active harvester connection'
+    );
+  });
+
+  test('should return an ApolloClient instance if harvester credentials are set', async () => {
```

Comment:
This test should just check if context.harvesterClient is equal to the instance passed to the constructor (no need to stub a query and call it